crates/transpiler/src/passes/basis_translator/mod.rs

```diff
@@ -620,19 +577,25 @@ fn replace_node(
                 .iter()
                 .map(|clbit| old_cargs[clbit.0 as usize])
                 .collect();
-            let new_op = match inner_node.op.view() {
-                OperationRef::Gate(gate) => gate.py_copy(py)?.into(),
-                OperationRef::Instruction(instruction) => instruction.py_copy(py)?.into(),
-                OperationRef::Operation(operation) => operation.py_copy(py)?.into(),
+            let new_op: PackedOperation = match inner_node.op.view() {
+                OperationRef::Gate(gate) => {
+                    Python::with_gil(|py| gate.py_copy(py).map(|op| op.into()))
+                        .map_err(|err| BasisTranslatorError::BasisGateError(err.to_string()))?
```

Comment:
Here you don't have the GIL anymore when calling `err.to_string()`, right? Then we also need to replace this with an hardcoded string error, right? Same in a bunch of other places 🙂

Reply:
Oh, since we're still going to use the GIL anyway. We can just move the message to be obtained inside of the GIL.